### ISSUE TITLE
improve code quality by CI linting pipelines

### ISSUE THREAD

**Issue by fembau** [research-software-directory/RSD-as-a-service#915]
Add a CI pipeline which calls ESLint on each commit.

Additionally use p.ex. https://prettier.io for the frontend and [focus](https://prettier.io/docs/en/why-prettier.html) on coding instead of formating. Prettier could also be integrated with ESLint.

**Comment by jmaassen** [comment]
Agreed. We should also do this for other languages (like the Java in the backend).

**Comment by fembau** [comment]
and SQL

**Comment by ewan-escience** [comment]
I'm trying out Prettier for Java and SQL files. You can see the differences here: https://github.com/research-software-directory/RSD-as-a-service/compare/915-code-quality?expand=1.

It's not all better, look for example at lines 43 -- 45 of authentication/src/main/java/nl/esciencecenter/rsd/authentication/OrcidLogin.java. It's apparently not good at handling if statements without curly brackets.

**Comment by ewan-escience** [comment]
Here is a screenshot of the code I meant in my previous comment, as the code will change:

![Screenshot_2023-08-07_11-17-00](https://github.com/research-software-directory/RSD-as-a-service/assets/91328544/61e53d2e-f6e1-4f1e-9923-25b456a1ea9f)

I will be trying out other formatters/linters, e.g. [Checkstyle](https://checkstyle.org/) for Java and [SQLFluff](https://sqlfluff.com/) for the database.

**Comment by dmijatovic** [comment]
For the frontend I find current eslint sufficient. It runs during the build and can be runned locally from the frontend folder. I do lint check regularly manually. We can add frontend linting check to CI though (`yarn lint`).

Adding Prettier to the chain requires additional dependency and VS plugin to be installed by fe developers. I would prefer lighter approach with eslint only.

**Comment by fembau** [comment]
Prettier could be [integrated with ESLint](https://prettier.io/docs/en/integrating-with-linters.html). I used this setup before. No extras for the IDE necessary, ESLint is enough.

Using prettier is very comfortable. Currently I have to switch it off because when making changes of course to whole file gets reformated and this makes it hard to understand PRs.

**Comment by fembau** [comment]
I found an issue in my VSCode setup and now ESLint is working as expected :)
Also I understood now that ESLint is used as formater.
